src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -2058,57 +2058,81 @@ void CodeGen::genFMAIntrinsic(GenTreeHWIntrinsic* node)
+    // Ideally we'd actually swap the operands in lsra and simplify codegen
+    // but its a bit more complicated to do so for many operands as well
+    // as being complicated to tell codegen how to pick the right instruction
+
+    instruction ins = INS_invalid;
+
     if (op1->isContained() || op1->isUsedFromSpillTemp())
     {
+        // targetReg == op3NodeReg or targetReg == ?
+        // op3 = ([op1] * op2) + op3
+        // 231 form: XMM1 = (XMM2 * [XMM3]) + XMM1
+        ins = _231form;
+        std::swap(emitOp1, emitOp3);
+
         if (targetReg == op2NodeReg)
         {
-            std::swap(emitOp1, emitOp2);
             // op2 = ([op1] * op2) + op3
             // 132 form: XMM1 = (XMM1 * [XMM3]) + XMM2
             ins = _132form;
-            std::swap(emitOp2, emitOp3);
+            std::swap(emitOp1, emitOp2);
         }
-        else
+    }
+    else if (op3->isContained() || op3->isUsedFromSpillTemp())
+    {
+        // targetReg could be op1NodeReg, op2NodeReg, or not equal to any op
+        // op1 = (op1 * op2) + [op3] or op2 = (op1 * op2) + [op3]
+        // ? = (op1 * op2) + [op3] or ? = (op1 * op2) + op3
+        // 213 form: XMM1 = (XMM2 * XMM1) + [XMM3]
+        ins = _213form;
+
+        if (!copiesUpperBits && (targetReg == op2NodeReg))
         {
-            // targetReg == op3NodeReg or targetReg == ?
-            // op3 = ([op1] * op2) + op3
-            // 231 form: XMM1 = (XMM2 * [XMM3]) + XMM1
-            ins = _231form;
-            std::swap(emitOp1, emitOp3);
+            // op2 = (op1 * op2) + [op3]
+            // 213 form: XMM1 = (XMM2 * XMM1) + [XMM3]
+            std::swap(emitOp1, emitOp2);
         }
     }
     else if (op2->isContained() || op2->isUsedFromSpillTemp())
     {
+        // targetReg == op1NodeReg or targetReg == ?
+        // op1 = (op1 * [op2]) + op3
+        // 132 form: XMM1 = (XMM1 * [XMM3]) + XMM2
+        ins = _132form;
+        std::swap(emitOp2, emitOp3);
+
         if (!copiesUpperBits && (targetReg == op3NodeReg))
         {
             // op3 = (op1 * [op2]) + op3
             // 231 form: XMM1 = (XMM2 * [XMM3]) + XMM1
             ins = _231form;
-            std::swap(emitOp1, emitOp3);
-        }
-        else
-        {
-            // targetReg == op1NodeReg or targetReg == ?
-            // op1 = (op1 * [op2]) + op3
-            // 132 form: XMM1 = (XMM1 * [XMM3]) + XMM2
-            ins = _132form;
+            std::swap(emitOp1, emitOp2);
         }
-        std::swap(emitOp2, emitOp3);
     }
     else
     {
-        // targetReg could be op1NodeReg, op2NodeReg, or not equal to any op
-        // op1 = (op1 * op2) + [op3] or op2 = (op1 * op2) + [op3]
-        // ? = (op1 * op2) + [op3] or ? = (op1 * op2) + op3
-        // 213 form: XMM1 = (XMM2 * XMM1) + [XMM3]
-        if (!copiesUpperBits && (targetReg == op2NodeReg))
+        // containedOpNum == 0
+        // no extra work when resultOpNum is 0 or 1
```

Comment:
remove the references of `containedOpNum` and `resultOpNum` and instead just explain that this is the case when no operand is contained and we pick the appropriate forms.

Reply:
I'm going to remove this in a follow up PR (will put it up about 5 minutes after merging).

This will allow us to get CI passing again without needing to wait on the PR to rerun all tests for a comment change.